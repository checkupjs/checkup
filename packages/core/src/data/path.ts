export function trimCwd(path: string, cwd: string) {
  return path.replace(`${cwd}/`, '');
}

export function trimAllCwd(paths: string[], cwd: string) {
  return paths.map((path) => trimCwd(path, cwd));
}
