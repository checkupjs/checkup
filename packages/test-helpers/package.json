{
  "name": "@checkup/test-helpers",
  "description": "Checkup's test helper library",
  "version": "0.0.10",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {
    "@checkup/core": "0.0.10",
    "fixturify-project": "^2.1.0",
    "handlebars": "^4.7.6",
    "stdout-stderr": "^0.1.13",
    "tmp": "^0.2.0"
  },
  "engines": {
    "node": ">= 12.*"
  },
  "files": [
    "/lib",
    "/static"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc --build",
    "test": "jest --passWithNoTests --no-cache"
  },
  "types": "lib/index.d.ts"
}
