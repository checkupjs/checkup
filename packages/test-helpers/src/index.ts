export { default as Project } from './ember-cli-fixturify-project';
export { stdout } from './stdout';
export { default as CheckupProject } from './checkup-fixturify-project';
export { default as Plugin } from './plugin';
