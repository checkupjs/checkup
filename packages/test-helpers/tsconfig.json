{
  "extends": "../../tsconfig-base.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "types": ["chai", "jest", "node"],
    "esModuleInterop": true
  },
  "references": [
    {
      "path": "../core"
    }
  ],
  "include": ["src/**/*"]
}
