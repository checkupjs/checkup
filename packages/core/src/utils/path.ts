import { isAbsolute, join, relative } from 'path';

export function toAbsolute(baseDir: string, filePath: string) {
  let absolutePath = filePath;

  if (!isAbsolute(filePath)) {
    absolutePath = join(baseDir, filePath);
  }

  return absolutePath;
}

export function toRelative(baseDir: string, filePath: string) {
  let relativePath = filePath;

  if (isAbsolute(relativePath)) {
    relativePath = relative(baseDir, filePath);
  }

  return relativePath;
}
