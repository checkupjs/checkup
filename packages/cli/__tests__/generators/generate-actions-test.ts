/* eslint-disable jest/expect-expect */

import * as helpers from 'yeoman-test';
import { resolve } from 'path';

import ActionsGenerator from '../../src/generators/actions';
import { generatePlugin } from '../__utils__/generator-utils';
import { testRoot } from '@checkup/test-helpers';

function assertActionsFiles(name: string, dir: string, extension: string = 'ts') {
  let root = testRoot(dir);

  expect(
    root.file(`${extension === 'ts' ? 'src' : 'lib'}/actions/${name}-actions.${extension}`).contents
  ).toMatchSnapshot();
}

describe('actions generator', () => {
  it('generates correct files with TypeScript', async () => {
    let baseDir = await generatePlugin();
    let dir = await helpers
      .run(ActionsGenerator, { namespace: 'checkup:actions' })
      .cd(baseDir)
      .withOptions({
        name: 'my-foo',
        path: '.',
      })
      .withPrompts({
        taskName: 'foo',
        typescript: true,
      });

    assertActionsFiles('my-foo', dir);
  });

  it('generates correct files with TypeScript in custom path', async () => {
    let baseDir = await generatePlugin({ path: './lib' });
    await helpers
      .run(ActionsGenerator, { namespace: 'checkup:actions' })
      .cd(resolve(baseDir, '../..'))
      .withOptions({
        name: 'my-foo',
        path: './lib/checkup-plugin-my-plugin',
      })
      .withPrompts({
        taskName: 'foo',
        typescript: true,
      });

    assertActionsFiles('my-foo', baseDir);
  });

  it('generates multiple correct files with TypeScript', async () => {
    let baseDir = await generatePlugin();
    let dir = await helpers
      .run(ActionsGenerator, { namespace: 'checkup:actions' })
      .cd(baseDir)
      .withOptions({
        name: 'my-foo',
        path: '.',
      })
      .withPrompts({
        taskName: 'foo',
        typescript: true,
      });

    dir = await helpers
      .run(ActionsGenerator, { namespace: 'checkup:actions' })
      .cd(baseDir)
      .withOptions({
        name: 'my-bar',
        path: '.',
      })
      .withPrompts({
        taskName: 'foo',
        typescript: true,
      });

    assertActionsFiles('my-foo', dir);
    assertActionsFiles('my-bar', dir);
  });

  it('generates correct files with JavaScript', async () => {
    let baseDir = await generatePlugin(
      { name: 'my-plugin', defaults: false },
      { typescript: false }
    );
    let dir = await helpers
      .run(ActionsGenerator, { namespace: 'checkup:actions' })
      .cd(baseDir)
      .withOptions({
        name: 'my-foo',
        path: '.',
      })
      .withPrompts({
        taskName: 'foo',
        typescript: false,
      });

    assertActionsFiles('my-foo', dir, 'js');
  });
});
