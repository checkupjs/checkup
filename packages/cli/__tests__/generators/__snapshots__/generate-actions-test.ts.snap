// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions generator generates correct files with JavaScript 1`] = `
"import { ActionsEvaluator } from '@checkup/core';

export function evaluateActions(taskResult, taskConfig) {
  let actionsEvaluator = new ActionsEvaluator();

  return actionsEvaluator.evaluate(taskConfig);
}"
`;

exports[`actions generator generates correct files with JavaScript 2`] = `
"let { getPluginName } = require('@checkup/core');

const {
  evaluateActions: evaluateMyFooActions
} = require('../actions/my-foo-actions');

module.exports = {
  register: function(args) {
    let pluginName = getPluginName(__dirname);

    args.register.actions('foo', evaluateMyFooActions);
  }
}
"
`;

exports[`actions generator generates correct files with TypeScript 1`] = `
"import { TaskConfig, ActionsEvaluator } from '@checkup/core';
import { Result } from 'sarif';

export function evaluateActions(taskResults: Result[], taskConfig: TaskConfig) {
  let actionsEvaluator = new ActionsEvaluator();

  return actionsEvaluator.evaluate(taskConfig);
}
"
`;

exports[`actions generator generates correct files with TypeScript 2`] = `
"import { RegistrationArgs, getPluginName } from '@checkup/core';

import { evaluateActions as evaluateMyFooActions } from '../actions/my-foo-actions';

export function register(args: RegistrationArgs) {
  let pluginName = getPluginName(__dirname);

  args.register.actions('foo', evaluateMyFooActions);
}
"
`;

exports[`actions generator generates correct files with TypeScript in custom path 1`] = `
"import { TaskConfig, ActionsEvaluator } from '@checkup/core';
import { Result } from 'sarif';

export function evaluateActions(taskResults: Result[], taskConfig: TaskConfig) {
  let actionsEvaluator = new ActionsEvaluator();

  return actionsEvaluator.evaluate(taskConfig);
}
"
`;

exports[`actions generator generates correct files with TypeScript in custom path 2`] = `
"import { RegistrationArgs, getPluginName } from '@checkup/core';

import { evaluateActions as evaluateMyFooActions } from '../actions/my-foo-actions';

export function register(args: RegistrationArgs) {
  let pluginName = getPluginName(__dirname);

  args.register.actions('foo', evaluateMyFooActions);
}
"
`;

exports[`actions generator generates multiple correct files with TypeScript 1`] = `
"import { TaskConfig, ActionsEvaluator } from '@checkup/core';
import { Result } from 'sarif';

export function evaluateActions(taskResults: Result[], taskConfig: TaskConfig) {
  let actionsEvaluator = new ActionsEvaluator();

  return actionsEvaluator.evaluate(taskConfig);
}
"
`;

exports[`actions generator generates multiple correct files with TypeScript 2`] = `
"import { TaskConfig, ActionsEvaluator } from '@checkup/core';
import { Result } from 'sarif';

export function evaluateActions(taskResults: Result[], taskConfig: TaskConfig) {
  let actionsEvaluator = new ActionsEvaluator();

  return actionsEvaluator.evaluate(taskConfig);
}
"
`;

exports[`actions generator generates multiple correct files with TypeScript 3`] = `
"import { RegistrationArgs, getPluginName } from '@checkup/core';

import { evaluateActions as evaluateMyBarActions } from '../actions/my-bar-actions';

import { evaluateActions as evaluateMyFooActions } from '../actions/my-foo-actions';

export function register(args: RegistrationArgs) {
  let pluginName = getPluginName(__dirname);

  args.register.actions('foo', evaluateMyFooActions);
  args.register.actions('foo', evaluateMyBarActions);
}
"
`;
